app/script/util/DebugUtil.js

```diff
@@ -58,6 +58,39 @@ z.util.DebugUtil = class DebugUtil {
       .then(() => this.logger.log(`Corrupted Session ID '${sessionId}'`));
   }
 
+  haveISentThisMessageToMyOtherClients(messageId) {
+    const userId = wire.app.repository.user.self().id;
+    const conversation = wire.app.repository.conversation.active_conversation();
+    const conversationId = conversation.id;
+    const clientId = wire.app.repository.client.currentClient().id;
+    const message = conversation.get_message_by_id(messageId);
+    const dateTime = new Date(message.timestamp());
+    let amountOfMessagesSent = 0;
+
+    const isOTRMessage = notification => notification.type === 'conversation.otr-message-add';
+    const isInCurrentConversation = notification => notification.conversation === conversationId;
+    const wasSentByOurCurrentClient = notification =>
+      notification.from === userId && (notification.data && notification.data.sender === clientId);
+    const hasExpectedTimestamp = notification => notification.time === dateTime.toISOString();
+
+    return wire.app.repository.event.notificationService
+      .getNotifications(undefined, undefined, 10000)
+      .then(({notifications}) =>
+        notifications
+          .map(notification => notification.payload)
+          .reduce((acc, payload) => acc.concat(payload))
+          .filter(isOTRMessage)
+          .filter(isInCurrentConversation)
+          .filter(wasSentByOurCurrentClient)
+          .filter(hasExpectedTimestamp)
```

Comment:
Shouldn't we merge these into a single filter function instead of touching every message 4 times?